File: evo.funders.commonmessages/v1/PHP/src/Models/MarketingPreference.php
<?php

namespace AzureFunderCommonMessages\PHP\Models;

use AzureFunderCommonMessages\PHP\Traits\JsonSerializer;

class MarketingPreference implements \JsonSerializable
{
	use JsonSerializer;
	protected bool $optInEmail;
	protected bool $optInSms;
	protected bool $optInPhone;
	protected bool $optInPost;

	public function getOptInEmail(): bool
	{
		return $this->optInEmail;
	}

	public function getOptInSms(): bool
	{
		return $this->optInSms;
	}

	public function getOptInPhone(): bool
	{
		return $this->optInPhone;
	}

	public function getOptInPost(): bool
	{
		return $this->optInPost;
	}

	public function setOptInEmail(bool $optInEmail): self
	{
		$this->optInEmail = $optInEmail;
		return $this;
	}

	public function setOptInSms(bool $optInSms): self
	{
		$this->optInSms = $optInSms;
		return $this;
	}

	public function setOptInPhone(bool $optInPhone): self
	{
		$this->optInPhone = $optInPhone;
		return $this;
	}

	public function setOptInPost(bool $optInPost): self
	{
		$this->optInPost = $optInPost;
		return $this;
	}
}

File: evo.funders.commonmessages/v1/DotNet/Models/MarketingPreference.cs
using Newtonsoft.Json;

namespace AzureFunderCommonMessages.DotNet.Models
{
    public class MarketingPreference
    {
        [JsonProperty("optInEmail", Required = Required.Always)]
        public bool OptInEmail { get; set; }

        [JsonProperty("optInPhone", Required = Required.Always)]
        public bool OptInPhone { get; set; }

        [JsonProperty("optInPost", Required = Required.Always)]
        public bool OptInPost { get; set; }

        [JsonProperty("optInSms", Required = Required.Always)]
        public bool OptInSms { get; set; }
    }
}